x/fungible/keeper/msg_server_update_gateway_contract.go

```diff
@@ -21,6 +23,16 @@ func (k msgServer) UpdateGatewayContract(
 		return nil, cosmoserrors.Wrap(authoritytypes.ErrUnauthorized, err.Error())
 	}
 
+	// parse the new gateway address
+	gatewayAddr := ethcommon.HexToAddress(msg.NewGatewayContractAddress)
```

Comment:
Would be cool to have `addr, ok := ethcommon.HexToAddress(...)` signature instead

Reply:
Sounds a good idea, we're using this method in many place so would be better to make the change once
Create an issue for it: https://github.com/zeta-chain/node/issues/2789